builders/pallets-precompiles/precompiles/gmp.md

```diff
@@ -0,0 +1,155 @@
+
+The most common instance that someone will have to interact with the precompile is in the case of a recovery, where a relayer doesn’t complete an MRL transaction. A user would have to find the VAA that comes with their origin chain transaction and then manually invoke the wormholeTransferERC20 function.
+
+## Building the Payload for Wormhole
+
+Currently the GMP precompile only supports sending liquidity through Wormhole.  
+
+To send liquidity, on the origin chain, users must invoke the [`transferTokensWithPayload` method](https://book.wormhole.com/technical/evm/tokenLayer.html#contract-controlled-transfer){target=_blank} on the [WormholeTokenBridge smart contract](https://github.com/wormhole-foundation/wormhole/blob/main/ethereum/contracts/bridge/interfaces/ITokenBridge.sol){target=_blank}. It requires a bytes payload, which must be formatted as a SCALE encoded MultiLocation object wrapped within another versioned type.  
+
+You may be unfamiliar with both SCALE encoding and MultiLocations if you are not familiar with the Polkadot ecosystem. [SCALE encoding](https://docs.substrate.io/reference/scale-codec/){target=_blank} is a compact form of encoding that Polkadot uses. The [MultiLocation type](https://wiki.polkadot.network/docs/learn-xcvm){target=_blank} is used to define a relative point in Polkadot, such as a specific account on a specific parachain (Polkadot blockchain).  
+
+Moonbeam’s GMP protocol requires a MultiLocation to represent the destination for liquidity routing, which most likely means an account on some other parachain. Whatever it is, this destination must be expressed as relative to Moonbeam.  
+
+!!! remember
+    MultiLocations being relative is important, because a parachain team may erroneously give you a MultiLocation relative to their own chain, which can be different. Providing an incorrect MultiLocation can result in **loss of funds**!   
+
+Each parachain will have their own methods of interpreting a Multilocation, and you will have to confirm with the project that the Multilocation that you form is correct. That being said, it is most likely that you will be forming a Multilocation with an account.
+
+There are multiple types of accounts that can be included in a MultiLocation, which you must know beforehand when constructing your MultiLocation. The two most common are:
+
+- **AccountKey20** — an account id that is 20-bytes in length, including Ethereum-compatible account ids such as those on Moonbeam
+- **AccountId32** — an account id that is 32-bytes in length, standard in Polkadot and its parachains
+
+In the following MultiLocation templates, replace `INSERT_PARACHAIN_ID` with the parachain ID of the network you wish to send funds to and replace `ADDRESS_HERE` with the address of the account you want to send funds to.  
```

Comment:
clearly state that this multilocation targets an account on another parachain, with moonbeam as the origin